invenio_rdm_records/services/components/pids.py

```diff
@@ -89,6 +90,29 @@ def publish(self, identity, draft=None, record=None):
             if record_id != draft_id:
                 changed_pids[scheme] = record_pids[scheme]
 
+        if current_app.config["DATACITE_ENABLED"]:
+            if "doi" not in draft_schemes:
+                draft_schemes.add("doi")
+            if "doi" not in draft_pids:
+                draft_pids["doi"] = {}
+
+            for scheme in draft_schemes:
```

Comment:
the amount of nested condititional statements makes it hard to untangle. Is it possible to simplify it?